pkg/upgrade/version_test.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package upgrade
+
+import (
+	. "github.com/pingcap/check"
+)
+
+func (t *testForEtcd) TestVersionJSON(c *C) {
+	v1 := NewVersion(1, "v2.0.0")
+	j, err := v1.toJSON()
+	c.Assert(err, IsNil)
+	c.Assert(j, Equals, `{"internal-no":1,"release-ver":"v2.0.0"}`)
+	c.Assert(j, Equals, v1.String())
+
+	v2, err := versionFromJSON(j)
+	c.Assert(err, IsNil)
+	c.Assert(v2, DeepEquals, v1)
+}
+
+func (t *testForEtcd) TestVersionFunc(c *C) {
+	var ve Version
+	c.Assert(ve.NotSet(), IsTrue)
+	c.Assert(CurrentVersion.NotSet(), IsFalse)
+
+	v1 := NewVersion(1, "v2.0.0")
+	v2 := NewVersion(2, "v2.1.0")
+	c.Assert(v1.Compare(v2), Equals, -1)
+	c.Assert(v2.Compare(v1), Equals, 1)
+
+	v11 := v1
+	c.Assert(v11.Compare(v1), Equals, 0)
```

Comment:
how about testing "we only compare `InternalNo` now."

Reply:
tested in https://github.com/pingcap/dm/pull/839/commits/862ad50bbf329a7b3ce8b6ed056f5232d702b222.